metric_learn/base_metric.py

```diff
@@ -602,7 +602,9 @@ def predict(self, triplets):
     prediction : `numpy.ndarray` of floats, shape=(n_constraints,)
       Predictions of the ordering of pairs, for each triplet.
     """
-    return np.sign(self.decision_function(triplets))
+    return np.array([-1 if (t <= 0) else 1 for t in
+                    self.decision_function(triplets)])
```

Comment:
How about just `2 * (self.decision_function(triplets) > 0) -1`?

Reply:
Just changed, this way is much simpler